File: capture-server-ui/src/app/sport-pip-capture/components/connection/connection-start/builders/live-event.ts
import { AbstractBuilder } from "app/blocks/strategies";
import { Data } from "app/sport-pip-capture/models/capture-interface";
import { ConnectionDetailsView, liveEventDetail } from "../views/live-event";
import { Transformer } from "app/blocks/transformer";

export class LiveEventBuilder extends AbstractBuilder<Data.LiveEventConnectionDetail, liveEventDetail> {
    compose(m: Data.LiveEventConnectionDetail, v: liveEventDetail) {
        v.title = m.title;
        v.detail.cityAddress = m?.detail?.cityAddress;
        v.detail.streetAdress = m?.detail?.streetAdress;
        v.detail.type = m?.detail?.type;
        v.id = m.id;
        v.dtEvent = m.dtEvent;
        v.level = m.level;
        v.program = m.program;
        v.sport = m.sport;
        v.year = m.year;
        v.venue.location = m?.venue?.location;
        v.time = m.time;
        v.type = m.type;
        Transformer.ComposeCollection(m?.connectionDetails, v?.connectionDetails, ConnectionDetailsBuilder)
    }
    decompose(v: liveEventDetail): Data.LiveEventConnectionDetail {
        return;
    }
    view(): liveEventDetail {
        return new liveEventDetail();
    }
}


export class ConnectionDetailsBuilder extends AbstractBuilder<Data.ConnectionDetails, ConnectionDetailsView> {
    compose(m: Data.ConnectionDetails, v: ConnectionDetailsView) {
        v.device = m.device;
        v.id = v.id;
        v.ipAddress = m.ipAddress;
        v.location = m.location;
        v.name = m.name;
        v.quality = m.quality;
        v.received = m.filesReceived;
        v.retries = m.retries;
        v.role = m.role;
        v.network=m.network;
        v.transmitStatus = m.transmitStatus;
    }
    decompose(v: ConnectionDetailsView): Data.ConnectionDetails {
        return;
    }
    view(): ConnectionDetailsView {
        return new ConnectionDetailsView()
    }

}

File: capture-server/src/live-event.cpp
#include "live-event.h"
#include "gateway.h"
#include "base.h"

LiveEvent::LiveEvent() {}

std::string LiveEvent::dtEvent()
{
    return m_model["dtEvent"].asString();
}

void LiveEvent::setDtEvent(const std::string &dtEvent)
{
    m_model["dtEvent"] = dtEvent;
}

std::string LiveEvent::level()
{
    return m_model["level"].asString();
}

void LiveEvent::setLevel(const std::string &level)
{
    m_model["level"] = level;
}

std::string LiveEvent::program()
{
    return m_model["program"].asString();
}

void LiveEvent::setProgram(const std::string &program)
{
    m_model["program"] = program;
}

std::string LiveEvent::sport()
{
    return m_model["sport"].asString();
}

void LiveEvent::setSport(const std::string &sport)
{
    m_model["sport"] = sport;
}

std::string LiveEvent::status()
{
    return m_model["status"].asString();
}

void LiveEvent::setStatus(const std::string &status)
{
    m_model["status"] = status;
}

std::string LiveEvent::title()
{
    return m_model["title"].asString();
}

void LiveEvent::setTitle(const std::string &title)
{
    m_model["title"] = title;
}

std::string LiveEvent::detailCityAddress()
{
    return m_model["detail"]["cityAddress"].asString();
}

void LiveEvent::setDetailCityAddress(const std::string &cityAddress)
{
    m_model["detail"]["cityAddress"] = cityAddress;
}

std::string LiveEvent::detailStreetAddress()
{
    return m_model["detail"]["streetAddress"].asString();
}

void LiveEvent::setDetailStreetAddress(const std::string &streetAddress)
{
    m_model["detail"]["streetAddress"] = streetAddress;
}

std::string LiveEvent::detailType()
{
    return m_model["detail"]["type"].asString();
}

void LiveEvent::setDetailType(const std::string &type)
{
    m_model["detail"]["type"] = type;
}

int LiveEvent::year()
{
    return m_model["year"].asInt();
}

void LiveEvent::setYear(int year)
{
    m_model["year"] = year;
}

std::string LiveEvent::venueLocation()
{
    return m_model["venue"]["location"].asString();
}

void LiveEvent::setVenueLocation(const std::string &location)
{
    m_model["venue"]["location"] = location;
}

int LiveEvent::time()
{
    return m_model["time"].asInt();
}

void LiveEvent::setTime(int time)
{
    m_model["time"] = time;
}

std::string LiveEvent::type()
{
    return m_model["type"].asString();
}

void LiveEvent::setType(const std::string &type)
{
    m_model["type"] = type;
}

std::string LiveEvent::countdown()
{
    return m_model["countdown"].asString();
}

void LiveEvent::setCountdown(const std::string &countdown)
{
    m_model["countdown"] = countdown;
}

std::vector<ConnectionDetail> LiveEvent::connectionDetails()
{
    std::vector<ConnectionDetail> connectionDetails;
    for (const auto &detail : m_model["connectionDetails"])
    {
        ConnectionDetail connectionDetail;
        connectionDetails.push_back(connectionDetail);
    }
    return connectionDetails;
}

void LiveEvent::setConnectionDetails(const std::vector<ConnectionDetail>& connectionDetails) {
    Json::Value details(Json::arrayValue);
    for (auto detail : connectionDetails) {
        Json::Value connectionDetail;
        connectionDetail["id"] = detail.id();
        connectionDetail["name"] = detail.name();
        connectionDetail["role"] = detail.role();
        connectionDetail["location"] = detail.location();
        connectionDetail["device"] = detail.device();
        connectionDetail["network"] = detail.network();
        connectionDetail["quality"] = static_cast<int>(detail.quality());
        connectionDetail["ipAddress"] = detail.ipAddress();
        connectionDetail["transmitStatus"] = static_cast<int>(detail.transmitStatus());
        connectionDetail["filesReceived"] = detail.filesReceived();
        connectionDetail["retries"] = detail.retries();
        details.append(connectionDetail);
    }
    m_model["connectionDetails"] = details;
}
